base/compiler/optimize.jl

```diff
@@ -374,86 +381,94 @@ isknowntype(@nospecialize T) = (T == Union{}) || isconcretetype(T)
 
 function statement_cost(ex::Expr, line::Int, src::CodeInfo, spvals::SimpleVector, slottypes::Vector{Any}, params::Params)
     head = ex.head
-    if is_meta_expr_head(head) || head == :copyast # not sure if copyast is right
+    if is_meta_expr_head(head)
         return 0
-    end
-    argcost = 0
-    for a in ex.args
-        if a isa Expr
-            argcost = plus_saturate(argcost, statement_cost(a, -1, src, spvals, slottypes, params))
-        end
-    end
-    if head == :return || head == :(=)
-        return argcost
-    end
-    extyp = line == -1 ? Any : src.ssavaluetypes[line]
-    if head == :call
+    elseif head === :call
         ftyp = argextype(ex.args[1], src, spvals, slottypes)
-        if isa(ftyp, Type)
-            return argcost
-        end
         if isa(ftyp, Const)
             f = (ftyp::Const).val
-            if isa(f, IntrinsicFunction)
-                iidx = Int(reinterpret(Int32, f::IntrinsicFunction)) + 1
-                if !isassigned(T_IFUNC_COST, iidx)
-                    # unknown/unhandled intrinsic
-                    return plus_saturate(argcost, params.inline_nonleaf_penalty)
-                end
-                return plus_saturate(argcost, T_IFUNC_COST[iidx])
+        elseif ftyp isa DataType && isdefined(ftyp, :instance)
+            f = ftyp.instance
```

Comment:
This branch should be dead-code

(also, it's available as the `singleton_type` function)

Reply:
This might indicate a problem elsewhere, but it's not dead code --- in fact I needed this to fix a regression in an earlier version of the PR. A call to `Core.sizeof` had `typeof(Core.sizeof)` as the type of the function.